src/pages/CreateCardPage/components/CardFieldContext/CardFieldReducer.ts

```diff
@@ -0,0 +1,37 @@
+import { CardField } from '@/types';
+
+type CardFieldActionTypes = 'UPDATE' | 'APPEND' | 'DELETE';
+
+export type CardFieldAction = {
+  type: CardFieldActionTypes;
+  payload: {
+    key: keyof CardField;
+    value: string;
+  };
+};
+
+const cardFieldReducer = (state: CardField, action: CardFieldAction) => {
+  const copyState = { ...state };
+
+  const nonKey = !(action.payload.key in copyState);
+  if (nonKey) throw new Error(`nonKey: ${action.payload.key}`);
+
+  switch (action.type) {
+    case 'UPDATE':
+      copyState[action.payload.key] = action.payload.value;
+      return copyState;
+    case 'APPEND':
+      copyState[action.payload.key] =
+        copyState[action.payload.key] + action.payload.value;
+      return copyState;
+
+    case 'DELETE':
+      copyState[action.payload.key] = copyState[action.payload.key].slice(
+        0,
+        -1
+      );
+      return copyState;
+  }
```

Comment:
```suggestion
  const hasKey = action.payload.key in state;
  if (!hasKey) throw new Error(`nonKey: ${action.payload.key}`); 

  switch (action.type) {
    case 'UPDATE':
      return { 
         ...state, 
         state[action.payload.key]: action.payload.value 
       } ;
   ...
  }
```

1. nonKey가 어떤 뜻인지 좀더 구체화를 해보면 좋을 것 같아요
2. 전개를 return할 때 하시면 좀 더 간결하게 코드가 구성돼요. copyState이긴하지만 객체를 직접 수정하는 동작은 지양해도 좋겠습니다.
3. 간단하게 immer와 비슷한 동작을 하는 함수를 만들어봐도 좋을 것 같아요

Reply:
3. 번에서 말씀하신 immer와 비슷하게 동작하는 함수는 아래와 같다고생각하는데 맞을까요?
```ts
const produce = <T,>(state: T, update : (draft : T ) => void) => {
  const copyState = { ...state };
  update(copyState);
  return copyState;
}
```
(일단 지금 상황에서는 depth 가 1인 객체를 카피하는 것이기 때문에 딥카피까지는 생각하지 않았습니다)

Reply:
[72c21e2](https://github.com/next-step/react-payments/pull/45/commits/72c21e2efe9c912def55051934f4bd75c97007c7)

Reply:
넵! 유틸함수로 분리되면 재사용하기도 편할 것 같아서 말씀드린부분이고 크게 의미는 없습니다!